### PR TITLE
Draft: rearrangment and splitting of Draft Edit + support for editing Part objects.

### PR DESCRIPTION
**Pull request by carlopav**
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [ ] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

File: src/Mod/Draft/draftguitools/gui_edit.py [hunk 1]
```diff
@@ -313,6 +315,18 @@ def proceed(self):
...
+    def numericInput(self, v, numy=None, numz=None):
+        """Execute callback by the toolbar to activate the update function.
+
+        This function gets called by the toolbar
+        or by the mouse click and activate the update function.
+        """
+        if numy:
+            v = App.Vector(v, numy, numz)
+        self.endEditing(self.obj, self.editing, v)
+        App.ActiveDocument.recompute()
```
Comment (vocx-fc) [inline src/Mod/Draft/draftguitools/gui_edit.py:327]: I haven't looked at the logic of this Draft Edit command with detail, but I wouldn't call the recompute in any function but the last one, that is, the finish method. But that's up to you.

File: src/Mod/Draft/draftguitools/gui_edit.py [hunk 2]
```diff
@@ -520,157 +529,59 @@ def endEditing(self, obj, nodeIndex, v=None):
...
+            self.trackers[obj.Name].append(trackers.editTracker(pos=points[ep], name=obj.Name, idx=ep))
 
     def resetTrackers(self, obj):
         """Reset Edit Trackers and set them again."""
         self.removeTrackers(obj)
         self.setTrackers(obj, self.getEditPoints(obj))
 
+    def resetTrackersBezier(self, obj):
+        # in future move tracker definition to DraftTrackers
+        from pivy import coin
```
Comment (vocx-fc) [inline src/Mod/Draft/draftguitools/gui_edit.py:563]: Import at the beginning of the file.

File: src/Mod/Draft/draftguitools/gui_edit.py [hunk 3]
```diff
@@ -904,920 +801,437 @@ def addPointToCurve(self, point, info=None):
...
+            return 
 
         doc = App.getDocument(str(node.documentName.getValue()))
-        self.obj = doc.getObject(str(node.objectName.getValue()))
-        if self.obj is None:
+        obj = doc.getObject(str(node.objectName.getValue()))
+        if obj is None:
             return
-        if not (Draft.getType(self.obj) in ["Wire", "BSpline", "BezCurve"]):
+        if not (utils.get_type(obj) in ["Wire", "BSpline", "BezCurve"]):
```
Comment (vocx-fc) [inline src/Mod/Draft/draftguitools/gui_edit.py:844]: This should be
```py
if something not in (...):
```

**Comment by vocx-fc** [comment]
Impressive. As you know, I haven't taken a look at the Draft Edit code much. As you heavily modified it since 6 months ago, as long as it worked, I didn't care. But it definitely makes sense to make it smaller and break it into smaller components.

**Comment by carlopav** [comment]
Well, this is just the first part of a bigger refactor: what I mainly did is to make the code more standard.
Now the code that handles EditTrackers creation and object updating almost completely work in object coordinate system in every splitted module. It's Draft_Edit main module that takes care of the transformation of the provided coordinates back and forth the global coordinate system, supporting better App::Part like objects. This should hopefully make easier to support link, but a bigger refactor of the selection system is needed.
Next step would probably be in the direction of better supporting the editing preview, that only works with few objects.
I also wanted to experiment a deeper integration of loo pivy graphics (the interaction code used by Curves wb), but it's defitely something that will take much time :)
About task panel, sorry, but it's not on my radar at the moment... i'd like to improve the interaction with the 3dView, and we can eventually take this into consideration after!

**Comment by carlopav** [comment]
thx!

File: src/Mod/Draft/draftguitools/gui_edit.py (hunk 1 above)
Comment (carlopav) [reply src/Mod/Draft/draftguitools/gui_edit.py:327]: Sorry to respond so late... 
This has always been the behavior of Draft_Edit... I think it's good to recompute the document after an EditTracker is moved: the object is updated so it's good to see how this reflects on other objects too... Perhaps in the future a "skip recompute until the end" option can be added.

**Pull request merged**